File: 백준/Bronze/1978. 소수 찾기/소수 찾기.py
import sys

n = int(sys.stdin.readline())
numbers = list(map(int, sys.stdin.readline().split()))

prime_count = 0

def is_prime(num):
    if num < 2:
        return False
    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            return False
    return True

for num in numbers:
    if is_prime(num):
        prime_count += 1

print(prime_count)

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    private static boolean isPrime(int num) {
        if (num < 2){
            return false;
        }
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        String[] numbers = br.readLine().split(" ");

        int primeCount = 0;

        for (int i = 0; i < N; i++){
            int num = Integer.parseInt(numbers[i]);

            if (isPrime(num)){
                primeCount++;
            }
        }
        System.out.println(primeCount);
    }


}
